core/embedders/embedders_jina.py

```
@@ -0,0 +1,1076 @@
+    
+    This embedder supports both:
+    1. Traditional embedding (for short texts, backward compatibility)
+    2. Late chunking (for long documents, superior context preservation)
+    """
+    
+    # Jina v4 constants
+    MAX_TOKENS = 32768  # Jina v4's context window
+    EMBEDDING_DIM = 2048
+    
+    def __init__(self, 
+                 device: str = None,
```

Ruff (0.12.2)

PEP 484 prohibits implicit `Optional`

Convert to `Optional[T]`

(RUF013)